### PR TITLE
doc: doc updates for 2.1.99-dev2

### PR DESCRIPTION
**Pull request by peknis**
Update versions, changelog and notes for the dev tag.

Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]
LGTM

File: doc/nrf/index.rst [hunk 1]
```diff
@@ -3,6 +3,23 @@
...
 
+.. important::
+
+   |NCS| v2.1.99-dev2 is a development tag and will be replaced by v2.2.0 in the future.
+
+   The development tag contains the following updates:
+
+   * New :ref:`wifi_scan_sample` sample.
+   * New :ref:`wifi_station_sample` sample.
+   * New :ref:`wifi_ficr_prog` sample, adding support for FICR/OTP programming.
```
Comment (ioannisg) [inline doc/nrf/index.rst:14]: I think here we need to list the radio test sample. This was not part of any previous release. the "FICR" Sample is only a part of this sample that support commands for FICR, OTP programming.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -241,12 +241,13 @@ Zigbee samples
 Wi-Fi samples
 -------------
 
-* Added :ref:`wifi_radio_test` sample with the radio test support.
-    * Added subcommands for FICR/OTP programming.
+* Added:
 
-* Added :ref:`wifi_scan_sample` sample that demonstrates how to scan for the access points.
+  * :ref:`wifi_radio_test` sample with the radio test support.
+    * Subcommands for FICR/OTP programming.
```
Comment (ioannisg) [inline doc/nrf/releases/release-notes-changelog.rst:247]: This reflects better what was added in this tag

**Review by ioannisg — CHANGES_REQUESTED** [review]
I see some inconsistencies between the changelog and the highlights. 
To my understanding there is no standalone FICR sample, rather an extension of the radio sample that adds some functionality. And since the Radio Test sample was not part of v2.1.0, it needs to be mentioned.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -241,12 +241,14 @@ Zigbee samples
...
 -------------
 
-* Added :ref:`wifi_radio_test` sample with the radio test support.
-    * Added subcommands for FICR/OTP programming.
+* Added:
+
+  * :ref:`wifi_radio_test` sample with the radio test support.
 
-* Added :ref:`wifi_scan_sample` sample that demonstrates how to scan for the access points.
+    * Subcommands for FICR/OTP programming.
```
Comment (b-gent) [inline doc/nrf/releases/release-notes-changelog.rst:248]: Incorrect indentation
image

**Review by b-gent — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -126,7 +126,7 @@ nRF5340 Audio
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:126]: Line 116: Missing colon
Line 118: CONFIG_SLM_DATAMODE_URC -> :ref:\`CONFIG_SLM_DATAMODE_URC <CONFIG_SLM_DATAMODE_URC>\`

Comment (peknis) [reply doc/nrf/releases/release-notes-changelog.rst:126]: Fixed.

**Comment by bama-nordic** [comment]
> thanks! @bama-nordic please do a final check on the last version

Looks good to me.

**Review by umapraseeda — APPROVED** [review]

**Comment by ioannisg** [comment]
@b-gent pls, revisit this one :)

**Review by b-gent — APPROVED** [review]

**Pull request merged**